src/uniswapMappings/uniswapPoolFactory.ts

```diff
@@ -0,0 +1,59 @@
+import { ALLOWED_TOKENS } from '../utils/pricing'
+/* eslint-disable prefer-const */
+import { ZERO_BI, ZERO_BD } from '../utils/constants'
+import { PoolCreated } from '../../generated/Factory/Factory'
+import { UniswapPool, Token, Bundle } from '../../generated/schema'
+import { Pool as PoolTemplate } from '../../generated/templates'
+import { log, Address } from '@graphprotocol/graph-ts'
+import { tokens } from '../modules'
+
+export function handlePoolCreated(event: PoolCreated): void {
+  // temp fix
+  if (event.params.pool == Address.fromHexString('0x8fe8d9bb8eeba3ed688069c3d6b556c9ca258248')) {
+    return
+  }
+
+  let bundle = Bundle.load('1')
+  if (bundle === null) {
+    // create new bundle for tracking eth price
+    let bundle = new Bundle('1')
+    bundle.ethPriceUSD = ZERO_BD
+    bundle.save()
+  }
+
+  let pool = new UniswapPool(event.params.pool.toHexString()) as UniswapPool
+  let token0 = tokens.getOrCreateToken(event.params.token0, ZERO_BI)
+  let token1 = tokens.getOrCreateToken(event.params.token1, ZERO_BI)
+
+    // bail if we couldn't figure out the decimals
+  if (token0.decimals == -1 || token1.decimals == -1) {
+    log.debug('mybug the decimal on some token was null', [])
```

Comment:
Is this code comming from Uniswap? maybe you can add more info on this? 

Sometimes we assume 18 decimals by default, I guess it would be not suitable here, right?

Reply:
YEah, it's comming from uniswap. We can assume 18 by default or we can add more info. PLMK what you prefer.